### PR TITLE
Payment w/ Stripe; router

### PR DESCRIPTION
**Pull request by zach-morris-txt**
This is a WIP for BE viewing. 
Will implement and submit LOOM video after meeting with BE team.

### CODE REVIEW

**Comment by AshleyCoder3** [comment]
Have questions about installations. will find out tomorrow

**Review by DanielReinagel — CHANGES_REQUESTED** [review]
Based on the code, you'd have to go to the route "/payment/payment" which seems somewhat redundant. Change that and this probably looks good.

**Review by jonahjtr — APPROVED** [review]
I love the consolidating and pretifying of the code

**Review by Ryan-E-Mark — APPROVED** [review]
Seconding what Daniel mentioned about the route being /payment/payment, not a big issue but could be a little cleaner. Other than that it looks like a great implementation of Stripe, good job getting this set up!

**Review by 19jlevitre — APPROVED** [review]
Looks good to me, possibly need to change how the error gets handled on line 36 of the striperouter file, but if logging it to the console is all we really need then looks great.

File: api/payment/stripeRouter.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+const express = require('express');
+const stripe = require('stripe')(process.env.STRIPE_KEY);
+const uuid = require('uuid');
+const router = express.Router();
+
+router.post('/payment', (req, res) => {
+  const { product, token } = req.body;
+  const idempotencyKey = uuid();
+
```
Comment (mcmillanj) [inline api/payment/stripeRouter.js:17]: router.post('/payment', (req,res) => {
const body ={ 
source: req.body.token.id,
amount: req.body.amount, //call amount when we generate request on front end
currency: 'USD'
}
})

File: api/payment/stripeRouter.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+//IMPLEMENTATION:
+//npm i stripe   ***not "stripe-checkout" but "stripe"***
+//import router into index
+//server router
+//.env file --> "STRIPE_KEY=" --> stripe test account public key
+
+// Code only need uuid installed??( line 15) and to be uncommented
+
+const express = require('express');
+const stripe = require('stripe')(process.env.STRIPE_KEY);
```
Comment (mcmillanj) [inline api/payment/stripeRouter.js:10]: //in server.js 
if( process.env.Node_ENV  !==  'production') require('dotenv').config();
const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);

**Review by mcmillanj — CHANGES_REQUESTED** [review]
made some suggestion towards this ,I am talking to paul about handling this

**Pull request merged**